File: UserServer/src/test/java/cc/nobrain/dev/userserver/member/MemberTest.java
/*
package cc.nobrain.dev.userserver.member;

import cc.nobrain.dev.userserver.domain.member.repository.MemberRepository;
import cc.nobrain.dev.userserver.domain.member.service.MemberService;
import cc.nobrain.dev.userserver.domain.member.service.dto.MemberDto;
import cc.nobrain.dev.userserver.domain.member.service.dto.MemberReq;
import org.junit.jupiter.api.*;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.transaction.annotation.Transactional;

@SpringBootTest
public class MemberTest {

    @Autowired
    private MemberService memberService;

    @Autowired
    private MemberRepository memberRepository;

    public static final String TEST_EMAIL = "nj.jo@test.com";
    public static final String TEST_PASSWORD = "!Aa123123";
    public static final String TEST_NAME = "nj.jo";

    @AfterEach
    @Transactional
    public void afterAll() {
        System.out.println("afterAll");
        System.out.println(memberRepository.deleteByEmail(TEST_EMAIL));
    }

    @Test
    @Order(1)
    @Transactional
    void register() {
        MemberReq.Register req = new MemberReq.Register(TEST_EMAIL, TEST_PASSWORD, TEST_NAME);
        MemberDto newMember = memberService.register(req);

        System.out.println(newMember);

        Assertions.assertEquals(newMember.getEmail(), TEST_EMAIL);
    }


    @Test
    @Order(2)
    void duplicate() {
        String email = "test@test.com";
        Assertions.assertEquals(memberService.duplicate(email), true);
    }


}
*/

File: UserServer/src/test/kotlin/cc/nobrain/dev/userserver/member/MemberTest.kt
package cc.nobrain.dev.userserver.member

import cc.nobrain.dev.userserver.common.component.RsaHelper
import cc.nobrain.dev.userserver.domain.member.entity.Member
import cc.nobrain.dev.userserver.domain.member.repository.MemberRepository
import cc.nobrain.dev.userserver.domain.member.service.MemberService
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest
import org.springframework.boot.test.autoconfigure.orm.jpa.TestEntityManager
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.AfterAll
import org.junit.jupiter.api.BeforeAll
import org.springframework.boot.test.context.SpringBootTest

@SpringBootTest
class MemberRepositoryTest {

    @Autowired
    lateinit var memberRepository: MemberRepository

    @Autowired
    lateinit var memberService: MemberService

    @Test
    fun setupTestAccount() {
        val testEmail = "test@test.com"
        val testPassword = "123123!"

        if (!memberRepository.existsByEmail(testEmail)) {
            val newMember = Member(
                email = testEmail,
                password = testPassword,
                name = "Test User",
                isVerified = true
            )
            memberRepository.save(newMember)
            println("Test account created: $testEmail")
        } else {
            println("Test account already exists: $testEmail")
        }

        val member = memberRepository.findByEmail(testEmail)
        assertThat(member).isNotNull
    }

    @Test
    fun deleteTestAccount() {
        val testEmail = "test@test.com"
        val member = memberRepository.findByEmail(testEmail)
        if (member != null) {
            memberRepository.delete(member)
            println("Test account deleted: $testEmail")
        } else {
            println("Test account not found: $testEmail")
        }
    }
}
